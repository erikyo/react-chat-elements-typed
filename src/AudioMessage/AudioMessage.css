.rce-mbox-audio {
    padding: 10px 0;
    max-width: 300px;
}

audio:focus {
    outline: none;
}
