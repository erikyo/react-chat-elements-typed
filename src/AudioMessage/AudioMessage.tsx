import type { FC } from "react";
import type { IAudioMessageProps } from "../types";
import "./AudioMessage.css";

const AudioMessage: FC<IAudioMessageProps> = (props) => {
	const controlsList = props.data?.controlsList ?? "nodownload";

	return (
		<div className={"rce-mbox-audio"} style={props.customStyle}>
			<audio {...props.audioProps} controls controlsList={controlsList}>
				<source
					src={props.data.audioURL}
					type={props.data?.audioType ?? "audio/mp3"}
				/>
				Your browser does not support the audio element.
			</audio>
			{props.text && <div className="rce-mbox-text">{props.text}</div>}
		</div>
	);
};

export default AudioMessage;
