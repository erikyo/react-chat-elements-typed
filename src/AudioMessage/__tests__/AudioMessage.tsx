import { describe, expect, it } from "vitest";
import React from "react";
import AudioMessage from "../AudioMessage.js";
import { render } from "@testing-library/react";

describe("AudioMessage component", () => {
	it("should render without issues", () => {
		const component = render(<AudioMessage data={{ audioURL: "test" }} />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
