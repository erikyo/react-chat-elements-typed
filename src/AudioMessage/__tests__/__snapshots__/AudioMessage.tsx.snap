// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AudioMessage component > should render without issues 1`] = `
<div>
  <div
    class="rce-mbox-audio"
  >
    <audio
      controls=""
      controlslist="nodownload"
    >
      <source
        src="test"
        type="audio/mp3"
      />
      Your browser does not support the audio element.
    </audio>
  </div>
</div>
`;
